Project-Descriptions-and-Plans/CV32E4_TBD/Project_Concept_for_CV32E4_TBD_Nov-27_2023.md

```diff
@@ -0,0 +1,250 @@
+* CV-X-IF support
+* CLIC support
+* Improved OBI interface support (bus error in particular)
+
+This new project will keep functionnal backward compatibility with CV32E40Pv2,
+but with relaxed rules compared to previous project. In particular, formal equivalence
+with previous design will not be required to make design work easier.
+
+Due to the strict backward compatibility rules of CV32E40Pv2, this new core is not
+named CV32E40Pv3 and a new naming should be found.
+
+## Summary of market or input requirements
+
+
+### Known market/project requirements at PC gate
+
+**Madatory features and design changes**
+
+Keep XPULP custom extensions that improve address manipulations and flow control:
+* Post-increment and register-register indexed load/store
+* Hardware loops
+* Immediate branch
+* Event load
+
+Introduce a dual opcode scheme for XPULP ALU instructions that have a strict RVB/RVP equivalent:
+* Multiply-accumulate
+* General ALU extensions
+* Bit manipulation
+* SIMD (16 and 8-bit data)
+note : this dual opcode scheme allows backard compatibility and open the door to smooth transition
+of existing programs from XPULP to RVB/RVP, if needed/relevant.
+
+Add support for RISC-V official bit manipulation instructions:
+* RVB zba (address manipulation) will be supported
+* RVB zbb (basic bit manipulations) will be supported
+* RVB zbs (single bit instructions) will be supported
+* RVB zbc (carry-less multiplications) : may be supported, to be confirmed if relevant for DSP applications
+* RVK zbkb (Bitmanip instructions for Cryptography) : pack, packh, packw, brev8, zip, unzip additionnal instructions may be supported, if relevant for DSP applications.
+* RVK zbkx (Crossbar permutation instructions) : xpem4 and xperm8 additionnal instructions may be supported, if relevant for DSP applications.
+* RVK zkn, zks, zkr and zkt are specific to cryptographic computations and will not be supported
+
+**Optionnal design changes**
+
+Introduce XPULPv3 custom extentions (preliminary, feature list may evolve):
+* ALU instructions that have a strict RVB/RVP equivalent opcode will be deprecated, new XPULPv3 opcodes will be RVB/RVP ones instead
+* Deprecate single operand *abs* instruction in favor of dual operand *pdif* (RVP superset of abs useful for manhattan distance computation)
+* Deprecate single operand *cv.cplxconj* instruction in favor of dual operand *pas* (RVP parallel add-sub)
+* Deprecate *cv.cplxmul.r* and *cv.cplxmul.i* in favor of a single instruction *cv.cplxmul* that computes both parts of complex multiplication
+* Introduce hermitian form of *cv.cplxmul* (rs1 x conj(rs2)) ? (widely used in DSP computation for telecom applications)
+* *list to be completed / updated*
+
+Add CV-X-IF interface:
+* End user product may require additionnal appliance-specific custom instructions, these instructions should be implemented in a co-processor.
+
+Floating Point Unit:
+* keep current CV-FPU integration or move it to CV-X-IF?
```

Comment:
I would keep it as it is verified

Reply:
I'm in line with you. But I don't plan to put any design effort on the FPU. We will keep CV32E40Pv2 RTL for this part of the design.
Is it OK for you ?

Reply:
```suggestion
* keep current CV-FPU integration unchanged as much as possible. If a design modification is envisioned, sequential equivalence with CV32E40Pv2 is required.
```